modules/tracing/mule-tracer-customization-impl/src/main/java/org/mule/runtime/tracer/customization/impl/provider/InitialSpanInfoIdentifier.java

```diff
@@ -23,14 +23,8 @@ public class InitialSpanInfoIdentifier {
   private final String stringId;
 
   public InitialSpanInfoIdentifier(Component component, String suffix, String overriddenName) {
-    // If we don't have the location we consider that this is different from
-    // any other initial span info, as it can be retrieved from different places.
-    if (component.getLocation() == null) {
-      this.stringId = randomUUID().toString();
-    } else {
-      this.stringId =
-          getLocationAsString(component.getLocation()) + "-" + defaultString(suffix) + "-" + defaultString(overriddenName);
-    }
+    this.stringId =
+        getLocationAsString(component.getLocation()) + "-" + defaultString(suffix) + "-" + defaultString(overriddenName);
```

Comment:
This will now fail with NPE if for any reason the component location is null, right? Maybe we should clarify that in the JDocs or even change the signature to something that requires a String with the location, so it is more clear that the null check has to be done beforehand.

Reply:
I'll change this so that the location is pass to the InitialSpanIdentifier